utils/filter_streams.h

```diff
@@ -109,6 +110,25 @@ class filter_ostream : public std::ostream {
     
     void close();
 
+    /* The next four overrides only take care of handling the deferred
+       exception flag, when an exception occurs. They otherwise execute their
+       std::ostream equivalent. */
+    template<typename T> filter_ostream & operator << (T && data)
+    {
+        try {
+            std::ostream & stdStream(*this);
+            stdStream << std::forward<T>(data);
```

Comment:
Ah you're right. Although the following is also valid:
`ostream::operator<<(std::forward<T>(data));`
The difference is that this will work even if the function is virtual. In any case, the difference is negligible in this case since I doubt the standard library will decide to make it virtual out of the blue.